### DISCUSSION TITLE
Punto 7.1.3

### DISCUSSION THREAD

**Discussion by Alejandro-A-DS** [rramosp/ai4eng.v1#129] [07 - DISEÑO DE ALGORITMOS]
Buenas noches:

Estoy realizando el tercer punto del taller 7.1 y estoy teniendo dificultades para optimizar la función loss. Dentro de esta función loss hay una multiplicación de la matriz X con el vector t, sin embargo al remplazar t por x0 parece que scipy entiende que x0 es un escalar y no un vector, pero no sé cómo hacer esa distinción al momento de optimizar. 

Muchas gracias de antemano y adjunto el código por si acaso.

`from scipy.optimize import minimize

sigm = lambda x: 1/(1+np.exp(-x))
loss_3 = lambda x0, x1: np.mean((np.apply_along_axis(sigm, 0, np.sum(X * x0, axis=1) + x1) - y) ** 2)

initx = np.random.random(size=2)*10-5

r = minimize(lambda x: loss_3(*x), method="BFGS", x0=initx)
r`

**Comment by Felipe-RA** [comment]
Hola Alejandro-666

Cuando compartas código porfa hazlo en formato de snippet de código para poderlo correr.

You can do it with the symbol ( `````` ) or clicking here:

<img width="689" alt="image" src="https://user-images.githubusercontent.com/49454068/170146713-5976069e-a7b0-4809-9391-6faa52729dfd.png">

Gracias!

**Reply by Alejandro-A-DS** [reply]
Sí señor, ya edité el post como lo sugeriste. Muchas gracias.

**Reply by miguelangelrestrepo** [reply]
Pana, ya lograste resolverlo?

**Reply by Felipe-RA** [reply]
@miguelangelrestrepo  Hola Miguel, este post ya es algo viejo. Voy a cerrarlo.

Porfa crea uno nuevo si tienes una pregunta específica con respecto a este punto

Gracias!
